File: algorithms/greedy/interval-scheduling/csharp/IntervalScheduling.cs
using System;
using System.Linq;

public class IntervalScheduling
{
    public static int Schedule(int[] arr)
    {
        int n = arr[0];
        var intervals = new (int start, int end)[n];
        for (int i = 0; i < n; i++)
        {
            intervals[i] = (arr[1 + 2 * i], arr[1 + 2 * i + 1]);
        }

        Array.Sort(intervals, (a, b) => a.end.CompareTo(b.end));

        int count = 0;
        int lastEnd = -1;
        foreach (var iv in intervals)
        {
            if (iv.start >= lastEnd)
            {
                count++;
                lastEnd = iv.end;
            }
        }

        return count;
    }
}

File: algorithms/greedy/interval-scheduling/java/IntervalScheduling.java
import java.util.Arrays;

public class IntervalScheduling {

    public static int intervalScheduling(int[] arr) {
        int n = arr[0];
        int[][] intervals = new int[n][2];
        for (int i = 0; i < n; i++) {
            intervals[i][0] = arr[1 + 2 * i];
            intervals[i][1] = arr[1 + 2 * i + 1];
        }

        Arrays.sort(intervals, (a, b) -> a[1] - b[1]);

        int count = 0;
        int lastEnd = -1;

        for (int[] interval : intervals) {
            if (interval[0] >= lastEnd) {
                count++;
                lastEnd = interval[1];
            }
        }

        return count;
    }
}
